### PR TITLE
Added support for selecting items with a selection rectangle

### LINKED ISSUE #94: Add a selection rectangle
This is standard function for all Microsoft apps.

### LINKED ISSUE #1731: Drag&Drop not working as intended
**Is your feature request related to a problem? Please describe.**
Not related to a problem

**Describe the solution you'd like**
video
Capture

When I try to move the folder "test" inside another folder, I have to point the folder (C:\ in this case). But Files UWP should allow drag&drop inside the red box on the screenshot. 

**Additional context**
Windows 10 Version 1909, Files UWP 0.13.0.0

### PR DESCRIPTION
**Pull request by gave92**
Closes #94
Fixes #1731

This PR adds support for drag selection for both GridViewBrowser and GenericFileBrowser.
I'm sure there are  bugs, please help me find them xD

Issues:
- ~Items loose focus when trying to rename them~
- ~Ctrl/Shift + click selection in ListView is broken~
- It's difficult to trigger the selection rectangle in GridView unless you click outside an item
- ~It's difficult to trigger the selection rectangle in ListView unless you click inside an item~
- ~Selecting multiple items prevents dragging the items to a different directory~
- ~Selecting one or more items and double clicking the DataGrid's empty space will result in the items being opened and the app sometimes crashes after doing so.~
- ~Deselecting DataGridRows via clicking the empty space sometimes isn't working unless I move the pointer to a different location~
- ~Invoking right click on multiple selected GridViewItems deselects every item except the one I had the pointer over~

Screenshot (11)
Screenshot (12)

### CODE REVIEW

**Comment by Jaiganeshkumaran** [comment]
This is really great!

**Comment by yaichenbaum** [comment]
@gave92 I test your changes and they seem to work well! I did notice a small regression when trying to rename files that they would lose focus before I could change the name.

**Comment by gave92** [comment]
Glad you like this xD
@yaichenbaum this in GridView, ListView or both?
Also I think I broke Cntrl/Shift + click selection in ListView

**Comment by yaichenbaum** [comment]
@gave92 Just in list view.

**Comment by gave92** [comment]
Some of the issue reported should now be fixed.

@yair100 Why "is it difficult to trigger the selection rectangle in ListView unless you click inside an item"?
Do you refer to the fact that you can't start a selection clicking on the empty space at the bottom of the ListView?

**Comment by yair100** [comment]
> @yair100 Why "is it difficult to trigger the selection rectangle in ListView unless you click inside an item"?
Do you refer to the fact that you can't start a selection clicking on the empty space at the bottom of the ListView?

Yes.

**Review by yair100 — DISMISSED** [review]

File: Files/Views/LayoutModes/GenericFileBrowser.xaml [hunk 1]
```diff
@@ -594,6 +594,8 @@
 
         <controls:DataGrid
             x:Name="AllView"
+            Background="Transparent"
+            VerticalAlignment="Stretch"
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GenericFileBrowser.xaml:598]: Will this prevent right clicking to get the current directories right click context menu?

Comment (gave92) [reply Files/Views/LayoutModes/GenericFileBrowser.xaml:598]: With this both item context menu and "empty space" context menu will still work

**Review by tsvietOK — DISMISSED** [review]

**Review by lukeblevins — CHANGES_REQUESTED** [review]
These changes are excellent! I hate to be that person who requests changes, but there are a couple bugs I found during testing:

- Selecting one or more items and double clicking the DataGrid's empty space will result in the items being opened and the app sometimes crashes after doing so.

- Deselecting DataGridRows via clicking the empty space sometimes isn't working unless I move the pointer to a different location

- Invoking right click on multiple selected GridViewItems deselects every item except the one I had the pointer over

**Review by lukeblevins — APPROVED** [review]
LGTM

**Review by yair100 — APPROVED** [review]

**Pull request merged**